### PR TITLE
Week 06 day 03 drill

### PR DESCRIPTION
**Pull request by cameronjacoby**

### CODE REVIEW

File: week-06-intro-ruby-on-rails/day-03/drill/readme.md [hunk 1]
```diff
@@ -1 +1,78 @@
...
+
+## Stretch Challenge
+
+* Implement a recursive solution (instead of iterative).
+
+## Resources
+
+* <a href="https://en.wikipedia.org/wiki/Binary_search_algorithm" target="_blank">Binary search algorithm - Wikipedia</a>
+* <a href="http://bigocheatsheet.com" target="_blank">Big-O Cheat Sheet</a>
+* <a href="http://www.cs.cmu.edu/~15110-f12/Unit05PtB-handout.pdf" target="_blank">Binary Search - Carnagie Mellon University</a>
```
Comment (eerwitt) [inline week-06-intro-ruby-on-rails/day-03/drill/readme.md]: **_Carnegie**_ Mellon University

File: week-06-intro-ruby-on-rails/day-03/drill/solutions.md [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+# Output:
+# return index of target if target is found
+# return false if target is not present
+
+# Pseudocode:
+# Find the low, high, and mid indexes of the input array.
+# Keep looping while the low index is less than the high index.
+# If the target value is equal to the value at the mid index, return the mid index.
+# If the target value is greater than value at the mid index:
+  # => Set the low index to the mid index + 1.
```
Comment (eerwitt) [inline week-06-intro-ruby-on-rails/day-03/drill/solutions.md]: Intentional indentation?

File: week-06-intro-ruby-on-rails/day-03/drill/solutions.md [hunk 3]
```diff
@@ -0,0 +1,87 @@
...
+  # => Set the mid index to the sum of the high index and the low index, divided by two.
+# Else if the target value is less than value at the mid index:
+  # => Set the high index to the mid index - 1.
+  # => Set the mid index to the sum of the high index and the low index, divided by two.
+
+# binary search iterative solution
+def binary_search_iterative(array, target)
```
Comment (eerwitt) [inline week-06-intro-ruby-on-rails/day-03/drill/solutions.md:26]: May consider using RDoc for comments but these look fine.

**Comment by eerwitt** [comment]
:+1: looks good, added a few comments.

Small note, it may be worth introducing the term `set` vs `array` in this drill.

**Comment by cameronjacoby** [comment]
@eerwitt Ok, typo and comments are all fixed up! Also just ran the solution code, and it's finding the first number if there are duplicates. Good call to check that!

File: week-06-intro-ruby-on-rails/day-03/drill/readme.md [hunk 4]
```diff
@@ -1 +1,78 @@
...
+# true
+# search continues on lower half
+
+array[mid] < target
+#=> 5 < 1
+#=> false
+# this condition won't occur because it's false
+```
+
+The process continues, eliminating half of the elements, then comparing the target value to the value of the middle element of the remaining elements, until the target value is either found (and its position is returned), or until the entire array has been searched (and `false` is returned).
```
Comment (eerwitt) [inline week-06-intro-ruby-on-rails/day-03/drill/readme.md:54]: Maybe a visual would help in explaining this?

**Comment by eerwitt** [comment]
:+1:

File: week-06-intro-ruby-on-rails/day-03/drill/readme.md (hunk 4 above)
Comment (cameronjacoby) [inline week-06-intro-ruby-on-rails/day-03/drill/readme.md:54]: @eerwitt Good call - just added it in!

**Pull request merged**